src/components/RecipeComponent.tsx

```diff
@@ -0,0 +1,65 @@
+import { Button, GridItem, Image, Text } from "@chakra-ui/react";
+import { AddIcon } from "@chakra-ui/icons";
+import { useMyStore } from "./store";
+import { NewRecipe } from "./types";
+import { useGetIngredientIds } from "./useGetIngredientsIds.tsx";
+import { useGetRecipePrice } from "./useGetRecipePrice.tsx";
+
+type Props = {
+  recipe: NewRecipe;
+};
+
+const RecipeComponent: React.FC<Props> = ({ recipe }) => {
+  const { recipesInCart, addRecipeToCart } = useMyStore();
+  const ingredientData = useGetIngredientIds(recipe);
+  const { totalPrice, productIds } = useGetRecipePrice(ingredientData, recipe);
```

Comment:
nit: proc rovnou neziskat ty `ingredientData` ktery potrebujem jen pro `useGetRecipePrice` v `useGetRecipePrice` primo?

Reply:
ted to nepouzivam, ale ty ingredientdata pak budu pouzivat pri hazeni do kosiku atd... nebo zatim predpokladam, pokud nakonec vyjde najevo, ze neni potreba, presunu to vse do usegetrecipeprice